sdk/core/azure-core/azure/core/pipeline/transport/_adapters.py

```diff
@@ -0,0 +1,607 @@
+from requests.adapters import BaseAdapter
+from urllib3.poolmanager import PoolManager, proxy_from_url
+from urllib3.response import HTTPResponse
+from urllib3.util import parse_url
+from urllib3.util import Timeout as TimeoutSauce
+from urllib3.util.retry import Retry
+from urllib3.exceptions import (
+    ClosedPoolError,
+    ConnectTimeoutError,
+    HTTPError as _HTTPError,
+    MaxRetryError,
+    NewConnectionError,
+    ProxyError as _ProxyError,
+    ProtocolError,
+    ReadTimeoutError,
+    SSLError as _SSLError,
+    ResponseError,
+    LocationValueError,
+)
+
+try:
+    from urllib3.contrib.socks import SOCKSProxyManager
+except ImportError:
+    def SOCKSProxyManager(*args, **kwargs):
+        raise InvalidSchema("Missing dependencies for SOCKS support.")
+
+_ver = sys.version_info
+is_py3 = (_ver[0] == 3)
+if is_py3:
+    basestring = (str, bytes)
+
+DEFAULT_POOLBLOCK = False
+DEFAULT_POOLSIZE = 10
+DEFAULT_RETRIES = 0
+DEFAULT_POOL_TIMEOUT = None
+
+def to_native_string(string, encoding='ascii'):
+    """Given a string object, regardless of type, returns a representation of
+    that string in the native string type, encoding and decoding where
+    necessary. This assumes ASCII unless told otherwise.
+    """
+    if isinstance(string, str):
+        out = string
+    else:
+        out = string.encode(encoding)
+
+    return out
+
+def _basic_auth_str(username, password):
+    """Returns a Basic Auth string."""
+
+    # "I want us to put a big-ol' comment on top of it that
+    # says that this behaviour is dumb but we need to preserve
+    # it because people are relying on it."
+    #    - Lukasa
+    #
+    # These are here solely to maintain backwards compatibility
+    # for things like ints. This will be removed in 3.0.0.
+    if not isinstance(username, basestring):
+        warnings.warn(
+            "Non-string usernames will no longer be supported in Requests "
+            "3.0.0. Please convert the object you've passed in ({!r}) to "
+            "a string or bytes object in the near future to avoid "
+            "problems.".format(username),
+            category=DeprecationWarning,
+        )
+        username = str(username)
+
+    if not isinstance(password, basestring):
+        warnings.warn(
+            "Non-string passwords will no longer be supported in Requests "
+            "3.0.0. Please convert the object you've passed in ({!r}) to "
+            "a string or bytes object in the near future to avoid "
+            "problems.".format(type(password)),
+            category=DeprecationWarning,
+        )
+        password = str(password)
+    # -- End Removal --
+
+    if isinstance(username, str):
+        username = username.encode('latin1')
+
+    if isinstance(password, str):
+        password = password.encode('latin1')
+
+    authstr = 'Basic ' + to_native_string(
+        b64encode(b':'.join((username, password))).strip()
+    )
+
+    return authstr
+
+
+class HTTPAdapter(BaseAdapter):
```

Comment:
Is this adapter autogenerated code by swagger?

Reply:
No. This is a customized version of requests.adapters.HTTPAdapter. requests.adapters.HTTPAdapter does not accept kwargs in send method so we cannot specify block_size. In order to work around a performance issue in _socket, we want to be able to specify block_size when creating socket.